instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -832,74 +840,81 @@ def _get_otel_send(
-                        and "headers" in message
-                    ):
-                        custom_response_attributes = (
-                            collect_custom_headers_attributes(
-                                message,
-                                self.http_capture_headers_sanitize_fields,
-                                self.http_capture_headers_server_response,
-                                normalise_response_header_name,
-                            )
-                            if self.http_capture_headers_server_response
-                            else {}
+
+            status_code = None
+            if message["type"] == "http.response.start":
+                status_code = message["status"]
+            elif message["type"] == "websocket.send":
+                status_code = 200
+
+            if not self.exclude_send_span:
+                with self.tracer.start_as_current_span(
+                    " ".join((server_span_name, scope["type"], "send"))
+                ) as send_span:
+                    if callable(self.client_response_hook):
+                        self.client_response_hook(send_span, scope, message)
+
+                    if send_span.is_recording():
+                        if message["type"] == "http.response.start":
+                            expecting_trailers = message.get("trailers", False)
+                        send_span.set_attribute(
+                            "asgi.event.type", message["type"]
                         )
-                        if len(custom_response_attributes) > 0:
-                            server_span.set_attributes(
-                                custom_response_attributes
+                        if status_code:
+                            set_status_code(
+                                send_span,
+                                status_code,
+                                None,
+                                self._sem_conv_opt_in_mode,
                             )
-                if status_code:
-                    # We record metrics only once
-                    set_status_code(
-                        server_span,
-                        status_code,
-                        duration_attrs,
-                        self._sem_conv_opt_in_mode,
-                    )
-                    set_status_code(
-                        send_span,
-                        status_code,
-                        None,
-                        self._sem_conv_opt_in_mode,
-                    )
 
-                propagator = get_global_response_propagator()
-                if propagator:
-                    propagator.inject(
-                        message,
-                        context=set_span_in_context(
-                            server_span, trace.context_api.Context()
-                        ),
-                        setter=asgi_setter,
-                    )
+                        if (
```

Comment:
See as these are in the `if not self.exclude_send_span:` if block, are we not performing the below logic anymore if `self._exclude_send_span` is `True`?

Reply:
You're right, server span logic should always be executed. They were previously needlessly nested which threw me